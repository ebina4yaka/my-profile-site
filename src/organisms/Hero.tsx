import { ReactElement } from 'react'
import Image from 'next/image'
import github from '../svg/github'
import SvgButton from '../atoms/SvgButton'

export default function Hero(): ReactElement {
  return (
    <div className="bg-pink-light dark:bg-pink-dark">
      <div className="lg:container mx-auto flex flex-col md:flex-row items-center">
        <div className="w-full lg:w-1/2 lg:py-6 text-center py-6">
          <Image
            className="rounded-full"
            src="/icon.jpg"
            alt="Icon"
            loading="eager"
            width={200}
            height={200}
          />
        </div>
        <div
          className="sm:box-content flex flex-col w-full lg:w-1/2 justify-center
                   items-start pt-12 pb-12 sm:pb-24 px-6"
        >
          <p className="uppercase tracking-loose">profile</p>
          <h1 className="font-bold text-3xl my-4">ebina4yaka</h1>
          <p className="leading-normal mb-4">
            I&apos;m frontend engineer and pokemon trainer.
          </p>
          <a
            href="https://github.com/ebina4yaka"
            target="_blank"
            rel="noreferrer noopener"
          >
            <SvgButton svgPath={github}>Github</SvgButton>
          </a>
        </div>
      </div>
    </div>
  )
}
