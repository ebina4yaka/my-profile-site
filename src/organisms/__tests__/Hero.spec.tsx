import renderer from 'react-test-renderer'
import Hero from '../Hero'

jest.mock('../../atoms/SvgButton', () => 'MockSvgButton')

it('Snapshot test for Hero', () => {
  const tree = renderer.create(<Hero />).toJSON()
  expect(tree).toMatchSnapshot()
})
