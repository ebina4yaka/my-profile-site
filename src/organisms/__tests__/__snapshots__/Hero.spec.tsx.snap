// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Hero 1`] = `
<div
  className="bg-pink-light dark:bg-pink-dark"
>
  <div
    className="lg:container mx-auto flex flex-col md:flex-row items-center"
  >
    <div
      className="w-full lg:w-1/2 lg:py-6 text-center py-6"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjIwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          alt="Icon"
          className="rounded-full"
          decoding="async"
          src="/_next/image?url=%2Ficon.jpg&w=640&q=75"
          srcSet="/_next/image?url=%2Ficon.jpg&w=256&q=75 1x, /_next/image?url=%2Ficon.jpg&w=640&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "inherit",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="sm:box-content flex flex-col w-full lg:w-1/2 justify-center
                   items-start pt-12 pb-12 sm:pb-24 px-6"
    >
      <p
        className="uppercase tracking-loose"
      >
        profile
      </p>
      <h1
        className="font-bold text-3xl my-4"
      >
        ebina4yaka
      </h1>
      <p
        className="leading-normal mb-4"
      >
        I'm frontend engineer and pokemon trainer.
      </p>
      <a
        href="https://github.com/ebina4yaka"
        rel="noreferrer noopener"
        target="_blank"
      >
        <MockSvgButton
          svgPath="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
        >
          Github
        </MockSvgButton>
      </a>
    </div>
  </div>
</div>
`;
